recipes/hopsmon.rb

```diff
@@ -0,0 +1,79 @@
+#Generate and sign certificates for hopsmon
+hopsworks_ip = private_recipe_ip('hopsworks', 'default')
+hopsworks_https_port = 8182
+
+if node.attribute?('hopsworks')
+  if node['hopsworks'].attribute?('https') and node['hopsworks']['https'].attribute?('port')
+    hopsworks_https_port = node['hopsworks']['https']['port']
+  end
+end
+
+node.override['kube-hops']['pki']['ca_api'] = "#{hopsworks_ip}:#{hopsworks_https_port}"
+hopsmon_kube_certs_dir = "#{node['kube-hops']['monitoring']['certs-dir']}"
+
+directory hopsmon_kube_certs_dir do
+  owner node['kube-hops']['user']
+  group node['kube-hops']['user']
+  mode '0700'
+  action :create
+  not_if { ::File.directory?(node['kube-hops']['monitoring']['certs-dir']) }
+end
+
+kube_hops_certs 'hopsmon' do
+  path        hopsmon_kube_certs_dir
+  subject     "/CN=hopsmon"
+  not_if      { ::File.exist?("#{node['kube-hops']['monitoring']['cert-crt']}") }
+end
+
+bash 'create-hopsmon-user-in-kubernetes' do
+  user node['kube-hops']['user']
+  group node['kube-hops']['group']
+  environment ({ 'HOME' => ::Dir.home(node['kube-hops']['user']) })
+  retries 6
+  retry_delay 30
+  code <<-EOH
+      kubectl config set-credentials #{node['kube-hops']['monitoring']['user']}  --client-certificate=#{node['kube-hops']['monitoring']['cert-crt']} --client-key=#{node['kube-hops']['monitoring']['cert-key']}
+  EOH
+  not_if "kubectl config view | grep #{node['kube-hops']['monitoring']['user']}", :environment => { 'HOME' => ::Dir.home(node['kube-hops']['user']) }
+end
+
+remote_file "#{node['kube-hops']['monitoring']['kube-state-metrics-image-tar']}" do
+  source node['kube-hops']['monitoring']['kube-state-metrics-image-url']
+  owner node['kube-hops']['user']
+  group node['kube-hops']['group']
+  mode "0700"
+end
+
+kube_state_metrics_image = node['kube-hops']['monitoring']['kube-state-metrics-image-tar']
+bash 'load_kube_state_metrics_image' do
+  user 'root'
+  group 'root'
+  code <<-EOH
+      docker load < #{kube_state_metrics_image}
+  EOH
+end
```

Comment:
You can't load the image here, otherwise the nodes won't have it. All the other images are loaded here: https://github.com/logicalclocks/kube-hops-chef/blob/master/recipes/default.rb - the default recipe is executed on all the nodes, both master and workers.